compiler/rustc_middle/src/lint.rs

```diff
@@ -373,12 +386,12 @@ pub fn struct_lint_level<'s, 'd>(
         if let Level::Expect(_) = level {
             let name = lint.name_lower();
             err.code(DiagnosticId::Lint { name, has_future_breakage, is_force_warn: false });
-            decorate(LintDiagnosticBuilder::new(err));
+
+            decorate(&mut err);
```

Comment:
Why does decorate have a return vale, if it is being ignored?

Reply:
Could you add comments explaining that to all functions that take such a strangely typed closure? I spent 15min digging through the code to figure out what the return value does...

Reply:
yeah, sure xD

Reply:
Thanks. :)